### PR TITLE
EDU-768: Migrate legacy "Running [Workers] in Docker" content into TypeScript dev guide

### PR DESCRIPTION
**Pull request by djmagee**
## What does this PR do?

- Migrates content wholesale from the "Running in Docker" section of the legacy page "Production Deploy Checklist for TypeScript SDK" into the TypeScript SDK developer's guide.
- Fixes a broken link in the "Debugging" section of  the TypeScript SDK developer's guide.

## Notes to reviewers

This PR only migrates content. Reviewing and updating the content is out of scope and can be addressed separately.

### CODE REVIEW

**Review by flossypurse — APPROVED** [review]

File: docs-src/typescript/how-to-run-a-worker-on-docker.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+description: Workers based on the TypeScript SDK can be deployed and run as Docker containers.
+tags:
+  - developer-guide
+  - sdk
+  - typescript
+---
+
+Workers based on the TypeScript SDK can be deployed and run as Docker containers.
+
+At this moment, we recommend usage of Node.js 16.
```
Comment (mjameswh) [inline docs-src/typescript/how-to-run-a-worker-on-docker.md:14]: No more known issue on Node 18, but known issues for Node 20. Would recommend Node 18 at this point.

Comment (mjameswh) [reply docs-src/typescript/how-to-run-a-worker-on-docker.md:14]: Can update 16 => 18 on all Docker snippets in the document.

**Review by mjameswh — APPROVED** [review]
LGTM

**Comment by bergundy** [comment]
My main concern is that we don’t mention bundling here

**Pull request merged**